README.md

```diff
@@ -0,0 +1,453 @@
+
+# Avenger
+At the heart of the DSL of the software there are two constructors: **query** and **command**.
+
+## queries
+The [**`query`**](#query) function allows you to query your data source and get an object of type [**`CachedQuery`**](#CachedQuery) in return.
+It accepts two parameters, the first is a function with a [**`Fetch`**](#Fetch) signature that is used to retrieve data from your data source, the second is an object with the[**`Strategy`**](#Strategy) signature that will be used to decide if the data stored by **Avenger** is still relevant or needs to be refetched.
+
+Although important, `query` is a pretty low-level API and **Avenger** offers some convenient utils with a [**`StrategyBuilder`**](#StrategyBuilder) signature that you should prefer over it (unless you have very specific needs):
+
+- **refetch:** runs the fetch function every time the data is requested.
+- **expire:** when the data is requested, the fetch function is run only if data in the `Cache` is older than the expiration defined, otherwise the cached value is used.
+- **available:** when the data is requested, if a cached value is available it is always returned, otherwise the fetch function is run and the result stored in the `Cache` accordingly.
+
+All these utils ask you to specify custom [**`Setoid`**](https://github.com/gcanti/fp-ts/blob/master/docs/modules/Setoid.ts.md) instances that will be used to check if an input is already present in one of the `Chache`'s keys (if the check is successful `Avenger` will try to use that value, otherwise it will resort to the `Fetch` function) but you can (and should) use them together with one of the built-in implementations that automatically take care of it:
+- **queryShallow:** will use a `Setoid` instance that performs a shallow equality check to compare inputs.
+- **queryStrict:** will use a `Setoid` instance that performs a strict equality check to compare inputs.
+- **queryJSON:** will use a `Setoid` instance that performs a strict equality check after transforming the data via JSON stringification to compare inputs.
+
+Some examples will help clarify:
+```ts
+/*
+  this implementation will always re-run the `Fetch` function
+  even if valid cached data is already present
+  and use shallow equality to compare input
+*/
+const myQuery = queryShallow(fetchFunction, refetch);
+
+/*
+  this implementation will never run the `Fetch` function
+  unless no valid data is present in the Cache
+  and use strict equality to compare input
+*/
+const myQuery = queryStrict(fetchFunction, available);
+
+/*
+  this implementation will run the `Fetch` function only if no valid data is present in the Cache
+  or t > 10000 ms passed till the last time data was fetched
+  and use strict equality to compare input
+*/
+const myQuery = queryJSON(fetchFunction, expire(10000));
+```
+
+Each time the `Fetch` function is run with some `inputs`, those same `inputs` are used as a `key` to store the result obtained:
+```
```

Comment:
missing language `ts`

Reply:
this is actually pseudo lang not real `ts`

Reply:
👍